### DISCUSSION TITLE
Von Mises as variable

### DISCUSSION THREAD

**Discussion by sidharthsarmah** [idaholab/moose#27300] [Q&A Modules: Solid mechanics]
Hi @GiudGiud , I need to find the ratio of hydrostatic stress to Von Mises Stress at the nodes.
For the hydrostatic stress declared as variables, I used the following:
```
[Variables]
  [./sH]
  [../]
[]

[Kernels]
  [./ph_stress]
    type = ADMaterialPropertyValue
    prop_name = 'hydrostatic_stress'
    variable = 'sH'
  [../]
[]

[Materials]
  [./hydrostatic_prop]
    type = ADRankTwoInvariant
    rank_two_tensor = stress
    property_name = hydrostatic_stress
    invariant = Hydrostatic
  [../]
```

Do you suggest I can similarly find the VM stress as variables?
```
[Variables]
  [./VM]
  [../]
[]

[Kernels]
  [./ph_stress]
    type = ADMaterialPropertyValue
    prop_name = 'vonmises'
    variable = 'VM'
  [../]

  [./hydrostatic_prop]
    type = ADRankTwoInvariant
    rank_two_tensor = stress
    property_name = vonmises
    invariant = VonMisesStress
  [../]
[]
```

**Comment by Rahim-Habibi** [comment]
Hi, 
Hydrostatic and Von mises stresses are defined as Auxkernel, you can call the proper aux to have them, here (https://mooseframework.inl.gov/source/auxkernels/RankTwoScalarAux.html). 
Then, to have the ratio of them, you can define a axuvariable/auxkenel to do the division.

**Reply by GiudGiud** [reply]
if you store both stresses as auxiliary variables (for example with the ADMaterialPropertyValue), then 
You can use a ParsedAux to perform the division between two auxiliary variables.
